dynatrace_extension/sdk/extension.py

```diff
@@ -156,19 +156,20 @@ def __new__(cls):
             Extension._instance = super(__class__, cls).__new__(cls)
         return Extension._instance
 
-    def __init__(self) -> None:
+    def __init__(self, name: str = "") -> None:
         # do not initialize already created singleton
         if hasattr(self, "logger"):
             return
 
         self.logger = extension_logger
+        self.logger.name = name
 
         self.extension_config: str = ""
         self._feature_sets: dict[str, list[str]] = {}
 
         # Useful metadata, populated once the extension is started
-        self.extension_name = ""  # Needs to be set by the developer if they so decide
-        self.extension_version = ""
+        self.extension_name = name
+        self.extension_version = self.get_version()
```

Comment:
@radu-stefan-dt did you test this? This doesn't exist until we parse the arguments later